html/mycakeapp/src/Controller/ReservesController.php

```diff
@@ -33,32 +34,76 @@ public function initialize()
 
     public function seat()
     {
+        //URL直打ち対策
+        if (empty($_SESSION['schedule_id'])) {
+            return $this->redirect(['controller' => 'error']);
+        }
         if (!empty($_SESSION['seat'])) { //ticketアクションから戻ってきたときにキャンセルフラグを立てる
-            $seatReservation = $this->SeatReservations->find()
+            $ExReservation = $this->SeatReservations->find()
                 ->where([
                     'member_id' => $_SESSION['seat']['member_id'],
                     'schedule_id' => $_SESSION['seat']['schedule_id'],
                     'column_number' => $_SESSION['seat']['column_number'],
                     'record_number' => $_SESSION['seat']['record_number'],
                 ])
                 ->toArray();
-            $seatReservation[0]['is_cancelled'] = '1';
-            if (!($this->SeatReservations->save($seatReservation['0']))) {
+            $ExReservation[0]['is_cancelled'] = '1';
+            if (!($this->SeatReservations->save($ExReservation['0']))) {
                 $this->request->session()->delete('seat');
                 return $this->redirect(['controller' => 'error']);
             }
             $this->request->session()->delete('seat');
         }
+        //席予約情報を取得
+        $seatReservations = $this->SeatReservations->find()
+            ->where(['schedule_id' => $_SESSION['schedule_id']])
+            ->andWhere(['is_cancelled' => 0]);
+        foreach ($seatReservations as $seatReservation) {
+            $Reserved[] = $seatReservation['column_number'] . '-' . $seatReservation['record_number'];
+        }
+        //予約済み情報を取得
+        $Reservations = $this->ReservationDetails->find()
+            ->where(['schedule_id' => $_SESSION['schedule_id']])
+            ->andWhere(['is_cancelled' => 0]);
+        foreach ($Reservations as $Reservation) {
+            $Reserved[] = $Reservation['column_number'] . '-' . $Reservation['record_number'];
+        }
+        if (isset($Reserved)) {
+            $this->set(compact('Reserved'));
+        }
+        if ($this->request->isPost()) {
+            $seatNumber = explode("-", $_POST['seat']);
+            if (in_array($_POST['seat'], $Reserved)) {
+                return $this->redirect(['controller' => 'error']);
+            }
```

Comment:
※緊急修正ではありません
エラーに飛ばすなら、セッションは削除した方が良いと思います。
```suggestion
            if (in_array($_POST['seat'], $Reserved)) {
            　   $this->request->session()->delete('schedule_id');
                return $this->redirect(['controller' => 'error']);
            }
```